src/deferred.js

```diff
@@ -359,5 +363,22 @@ jQuery.extend( {
 	}
 } );
 
+if ( window.console && window.console.warn ) {
+
+	// These usually indicate a programmer mistake during development,
```

Comment:
How true is this? An obvious and common example is `JSON.parse` throwing SyntaxError, but there are also cases like `Date.prototype.toISOString` throwing RangeError. Promise examples often encourage developers to optimize for non-exceptional execution, and this could get noisy when they do something like:
```js
get( … ).then( assertSuccess ).then( JSON.parse ).then( handleResponse )
	.catch( handleError )
	.then( finish );
```